src/qt/intro.cpp

```diff
@@ -161,6 +161,37 @@ Intro::Intro(QWidget *parent, int64_t blockchain_size_gb, int64_t chain_state_si
         UpdateFreeSpaceLabel();
     });
 
+    bool have_user_assumevalid = false;
+    if (gArgs.IsArgSet("-assumevalid")) {
+        const auto user_assumevalid = gArgs.GetArg("-assumevalid", /* ignored default; determines return type */ "");
+        if (uint256S(user_assumevalid).IsNull()) {
+            // -assumevalid=0: default checkbox to off, and initialise with chainparams later
+            ui->assumevalid->setChecked(false);
+        } else {
+            // -assumevalid=blockhash: initialise with the user-specified value, enabled
+            ui->assumevalid->setChecked(true);
+            ui->assumevalidBlock->setText(QString::fromStdString(user_assumevalid));
+            have_user_assumevalid = true;
+        }
+    }
```

Comment:
nit: while not mandatory nor blocker, maybe you could consider refactoring these `if` statements into functions like the above `UpdateFreeSpaceLabel()`, as it could improve code readability and maintainability, especially as we keep adding more logic.